src/audio/module_adapter/module/volume/volume.c

```diff
@@ -281,6 +281,7 @@ static void volume_ramp(struct processing_module *mod)
 			cd->volume[i] = cd->tvolume[i];
 
 		cd->ramp_finished = true;
+		cd->copy_needed = true;
```

Comment:
I wonder if it nesssary here?

Reply:
I am not 100% sure also, this should be playback beginning, from ready to active, if for first time, then copy is needed, otherwise, the gain is not there, put it here for 100% correct volume.